### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by yankejustin**
<h1>Performance Changes</h1>
- Reduced some string concatenation
- Reduced a few decisions that are made very often in the program
- Removed an unnecessary method call

<h1>Code Quality Improvements</h1>
- Make sure the <code>MemoryStream</code> used when adding a Texture Pack is disposed of; even if an exception is thrown after its initialization.
- Ensure old and unused texture fonts are disposed of.

<h1>Other Notes</h1>
I know it doesn't seem like much, but a few more of these and it will be noticeably faster. :)

### CODE REVIEW

**Comment by SirCmpwn** [comment]
Thanks! FYI, the client is being rewritten under the opentk branch, so don't spend too much time optimizing the master branch's client. Will merge this once the travis build passes.

File: TrueCraft.API/Ray.cs [hunk 1]
```diff
@@ -94,18 +94,29 @@ public override int GetHashCode()
...
+                    maxT.Y = (box.Max.Y - Position.Y) / Direction.Y;
+            }
+
+            if (Direction.Z != 0.0f)
+            {
+                if (Position.Z < box.Min.Z)
+                    maxT.Z = (box.Min.Z - Position.Z) / Direction.Z;
+                else if (Position.Z > box.Max.Z)
+                    maxT.Z = (box.Max.Z - Position.Z) / Direction.Z;
+            }
```
Comment (SirCmpwn) [inline TrueCraft.API/Ray.cs]: Our Ray code is taken mostly from MonoGame, by the way. You might want to submit your improvements upstream if they're still relevant.

**Comment by SirCmpwn** [comment]
:tada:

File: TrueCraft.API/Ray.cs (hunk 1 above)
Comment (yankejustin) [inline TrueCraft.API/Ray.cs]: Got it. Thank you.

**Pull request merged**